### PR TITLE
feat(draw): New drawing tool

### PR DESCRIPTION
**Pull request by drekss**
Adding new drawing tool to igo2. It allows Point, Linestring, Polygon and Circle geometries to be drawn on the map.

- new "ngx-color-picker": "^9.1.0" dependency in pakage.json
- tool name: "draw"
- need to be add in toolbar and tools to be used

### CODE REVIEW

**Comment by pelord** [comment]
I suggest to change the tool icon. Instead of using the pencil icon, I suggest to use the draw icon.

image

File: packages/geo/src/lib/draw/draw/draw.component.ts [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+
+        this.layer = new VectorLayer({
+          title: this.languageService.translate.instant('igo.geo.draw.drawing'),
+          zIndex: 200,
+          source: new FeatureDataSource(),
+          style: (feature, resolution) => {
+            return this.drawStyleService.createDrawLayerStyle(feature, resolution, this.map, true);
+          },
+          showInLayerList: true,
+          exportable: false,
```
Comment (pelord) [inline packages/geo/src/lib/draw/draw/draw.component.ts:241]: You should let the layer being exportable. @drekss

Comment (pelord) [reply packages/geo/src/lib/draw/draw/draw.component.ts:241]: But there is an issue with the shapefile output...

Comment (pelord) [reply packages/geo/src/lib/draw/draw/draw.component.ts:241]: {"error":true,"msg":"ERROR 1: Attempt to write non-point (POLYGON) geometry to point shapefile.\nERROR 1: Unable to write feature 1 from layer ogr_57d2b442232.\nERROR 1: Terminating translation prematurely after failed\ntranslation of layer ogr_57d2b442232 (use -skipfailures to skip errors)\n"}

Comment (pelord) [reply packages/geo/src/lib/draw/draw/draw.component.ts:241]: Maybe split the features on export  by geometry type..
I.E.
if shapefile and multigeometryDraw:
    split by geomety and make a call to ogre for each geometry type.

File: packages/geo/src/lib/draw/draw/draw.component.ts [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+        this.toggleLabel = !this.toggleLabel;
+        this.changeStoreLayerStyle(this.toggleLabel);
+    }
+
+    private updateLabelOfOlGeometry(olGeometry: OlPoint | OlLineString | OlPolygon | OlCircle, label: string) {
+        olGeometry.setProperties({_label: label}, true);
+    }
+
+    private checkStoreCount() {
+        console.log(this.store.count$.getValue());
```
Comment (pelord) [inline packages/geo/src/lib/draw/draw/draw.component.ts:474]: remove this!

File: packages/geo/src/lib/draw/draw/draw.component.html [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+            [cpPosition]="'bottom'"
+            [cpOutputFormat] = "'rgba'"
+            [cpCancelButton]="true"
+            [cpCancelButtonText]="'igo.geo.draw.cancelColorPicker' | translate"
+            [value]="fillColor"
+            (colorPickerSelect)="changeStoreLayerStyle(toggleLabel)">
+        </mat-form-field>
+        <br>
+
+        {{'igo.geo.draw.stroke' | translate}} <mat-icon svgIcon="palette"></mat-icon> &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
```
Comment (mbarbeau) [inline packages/geo/src/lib/draw/draw/draw.component.html:56]: replace &nbsp  by css

File: packages/geo/src/lib/draw/draw/draw.component.ts (hunk 2 above)
Comment (mbarbeau) [reply packages/geo/src/lib/draw/draw/draw.component.ts:477]: :+1:

**Review by mbarbeau — CHANGES_REQUESTED** [review]

**Review by mbarbeau — APPROVED** [review]

**Pull request merged**